draft-ietf-quic-recovery.md

```diff
@@ -762,47 +762,95 @@ The minimum congestion window is the smallest value the congestion window can
 decrease to as a response to loss, ECN-CE, or persistent congestion.
 The RECOMMENDED value is 2 * max_datagram_size.
 
-## Slow Start
+## Congestion Control States
 
-While in slow start, QUIC increases the congestion window by the
-number of bytes acknowledged when each acknowledgment is processed, resulting
-in exponential growth of the congestion window.
+The congestion controller has three distinct states, as shown in
+{{fig-cc-fsm}}.
 
-QUIC exits slow start upon loss or upon increase in the ECN-CE counter.
-When slow start is exited, the congestion window halves and the slow start
-threshold is set to the new congestion window.  QUIC re-enters slow start
-any time the congestion window is less than the slow start threshold,
-which only occurs after persistent congestion is declared.
+~~~
+                 Start or         +------------+
+            persistent congestion |   Slow     |
+        (O)---------------------->|   Start    |
+                                  +------------+
+                                        |
+                                Loss or |
+                        ECN-CE increase |
+                                        v
+ +------------+     Loss or       +------------+
+ | Congestion |  ECN-CE increase  |  Recovery  |
+ | Avoidance  |------------------>|   Period   |
+ +------------+                   +------------+
```

Comment:
It would be nice if you drew what happens in Recovery Period in the case of  Loss or ECN-CE increase. In my understanding, Recovery Period continues and both cwin and ssthresh are not changed. I'm not sure if congestion_recovery_start_time is changed.

Reply:
Nothing happens.  The only real actions occur on entry (and maybe on exit, depending on when you decide to apply the congestion window decrease).  The text and pseudocode capture this; I had trouble fitting anything more into the diagram.

Reply:
OK.

Reply:
Shouldn't this also show how to transition out of recovery?

Reply:
Looks like that's been added.